src/Time/Rational.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE ExplicitNamespaces   #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Time.Rational
+       ( Rat
+       , type (%)
+       , DivRat
+       , Gcd
+       , Normalize
+       ) where
+
+import GHC.TypeNats
+
+-- | Data structure represents the rational number.
+data Rat = MkRat Nat Nat
+
+type (%) = 'MkRat
+infixl 7 %
+
+type family DivRat (a :: Rat) (b :: Rat) :: Rat where
+    DivRat (a % b) (c % d) = Normalize ((a * d) % (b * c))
+
+type family Gcd (a :: Nat) (b :: Nat) :: Nat where
```

Comment:
I think it's a good idea to add a couple Haddock examples to every type family with `:kind!`. Also, we probably can write some tests for these type families later on...